### ISSUE TITLE
ASK CLI Deploy issue

### ISSUE THREAD

**Issue by rakhi2104** [alexa/alexa-skills-kit-sdk-for-nodejs#489]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Expected Behavior
The regular `ask deploy` or `ask deploy -t lambda` worked fine till I removed the node_modules and reinstalled the packages today morning.

## Current Behavior
```
/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^

TypeError: Cannot read property 'slice' of undefined
    at Object.removeLATESTFromArn (/usr/local/lib/node_modules/ask-cli/lib/utils/lambda-operation/lambda-endpoint-utility.js:180:13)
    at _fillUpInfoFromResourcesConfig (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:141:65)
    at createLambda.createLambda (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:67:21)
    at Object.createLambda (/usr/local/lib/node_modules/ask-cli/lib/utils/lambda-operation/create-lambda.js:20:13)
    at installNodeModule.install (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:66:30)
    at async.each (/usr/local/lib/node_modules/ask-cli/lib/utils/install-node-module.js:41:9)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:473:16
    at eachOfArrayLike (/usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:1057:9)
    at eachOf (/usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:1117:5)
    at Object.eachLimit (/usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:3172:5)
    at Object.module.exports.install (/usr/local/lib/node_modules/ask-cli/lib/utils/install-node-module.js:37:11)
    at checkRuntimeForAskSdkUsage (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:65:31)
    at async.each (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:110:9)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:473:16
    at iteratorCallback (/usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:1064:13)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:969:16
    at retrieveLambda.retrieveLambdaRuntime (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:104:13)
    at getLambdaFunction (/usr/local/lib/node_modules/ask-cli/lib/utils/lambda-operation/retrieve-lambda.js:33:9)
    at Response.lambdaClient.getFunction (/usr/local/lib/node_modules/ask-cli/lib/utils/lambda-operation/retrieve-lambda.js:44:9)
    at Request.<anonymous> (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:364:18)
    at Request.callListeners (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
```

## Environment
* Ubuntu 18.04 LTS

## Node.js and NPM Info
* npm: v6.4.1
* node: v10.11.0

**Comment by tianrenz** [comment]
Forward to service team.

**Comment by RonWang** [comment]
Hey @rakhi2104 can I ask what's your CLI version? I think once you go to the latest version of CLI v1.5.1, you won't have this issue anymore! Let me know if that solves

**Comment by rakhi2104** [comment]
Hey @RonWang, the slice issue solved with v1.5.1, but a new issue came up.
```
/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^
TypeError: Cannot read property 'includes' of undefined
    at _mergeEndpointInfoToResource (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:146:47)
    at _preMergeLambda (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:41:33)
    at async.forEach (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:308:9)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:473:16
    at iteratorCallback (/usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:1064:13)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:969:16
    at async.forEach (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:305:13)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:473:16
    at iteratorCallback (/usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:1064:13)
    at /usr/local/lib/node_modules/ask-cli/node_modules/async/dist/async.js:969:16
    at retrieveLambda.retrieveLambdaFunction (/usr/local/lib/node_modules/ask-cli/lib/deploy/deploy-lambda.js:302:17)
    at getLambdaFunction (/usr/local/lib/node_modules/ask-cli/lib/utils/lambda-operation/retrieve-lambda.js:23:13)
    at Response.lambdaClient.getFunction (/usr/local/lib/node_modules/ask-cli/lib/utils/lambda-operation/retrieve-lambda.js:45:9)
    at Request.<anonymous> (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:364:18)
    at Request.callListeners (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/usr/local/lib/node_modules/ask-cli/node_modules/aws-sdk/lib/request.js:685:12)
```

**Comment by RonWang** [comment]
@rakhi2104 Hey that actually seems something is corrupted in your project config's Lambda resource list... 
After checking the logic, this will happen when you do have Lambda resource but that Lambda resource doesn't have an "alexaUsage" mapping for it, which is not true if this config is created by CLI. Have you manually modified that? To unblock, you can add back the alexaUsage, by putting the value as an array with one or multiple usage(s) in the format of "{skill-type}/{region}".

Thanks,
Ron

**Comment by rakhi2104** [comment]
Thanks for the suggestion @RonWang :+1: Instead of messing again with the local files, I cloned the project using ask-cli and successfully deployed it.
You can close this issue now. :100:

**Comment by tianrenz** [comment]
Thanks for the help! @RonWang 

Closing the issue now, please feel free to reopen if the problem persists
